connectors/src/connectors/confluence/temporal/activities.ts

```diff
@@ -294,7 +325,44 @@ export async function confluenceUpsertPageActivity({
 
   localLogger.info("Upserting Confluence page in DB.");
 
-  await upsertConfluencePageInDb(connectionId, dataSourceConfig, page);
+  await upsertConfluencePageInDb(connector.id, page, visitedAtMs);
+}
+
+export async function confluenceRemoveUnvisitedPagesActivity({
+  connectorId,
+  lastVisitedAt,
+  spaceId,
+}: {
+  connectorId: ModelId;
+  lastVisitedAt: number;
+  spaceId: string;
+}) {
+  const connector = await Connector.findOne({
+    where: {
+      id: connectorId,
+    },
+  });
+  if (!connector) {
+    throw new Error(`Connector not found (id: ${connectorId})`);
+  }
+
+  const unvisitedPages = await ConfluencePage.findAll({
+    attributes: ["pageId"],
+    where: {
+      connectorId,
+      spaceId,
+      lastVisitedAt: {
+        [Op.ne]: new Date(lastVisitedAt),
+      },
+    },
+  });
+
+  const dataSourceConfig = dataSourceConfigFromConnector(connector);
+
+  for (const page of unvisitedPages) {
+    // TODO(2024-01-22 flav) Add an extra check to ensure that the page does not exist anymore in Confluence.
```

Comment:
:+1: 

Also let's plan to add a production check that makes sure that all confluence pages in core are accounted for in connectors (some similar exist)

And maybe another production check to make sure all confluence workflows are running (some similar exist)